app/src/organisms/ProtocolSetupParameters/__tests__/ProtocolSetupParameters.test.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react'
+import { it, describe, beforeEach, vi, expect } from 'vitest'
+import { fireEvent, screen } from '@testing-library/react'
+import { i18n } from '../../../i18n'
+import { renderWithProviders } from '../../../__testing-utils__'
+import { ProtocolSetupParameters } from '..'
+import { useMostRecentCompletedAnalysis } from '../../LabwarePositionCheck/useMostRecentCompletedAnalysis'
+import { mockRunTimeParameterData } from '../../../pages/ProtocolDetails/fixtures'
+import type * as ReactRouterDom from 'react-router-dom'
+
+const mockGoBack = vi.fn()
+vi.mock('../../LabwarePositionCheck/useMostRecentCompletedAnalysis')
+vi.mock('react-router-dom', async importOriginal => {
+  const reactRouterDom = await importOriginal<typeof ReactRouterDom>()
+  return {
+    ...reactRouterDom,
+    useHistory: () => ({ goBack: mockGoBack } as any),
+  }
+})
+
+const render = (
+  props: React.ComponentProps<typeof ProtocolSetupParameters>
+) => {
+  return renderWithProviders(<ProtocolSetupParameters {...props} />, {
+    i18nInstance: i18n,
+  })
+}
+describe('ProtocolSetupParameters', () => {
+  let props: React.ComponentProps<typeof ProtocolSetupParameters>
+
+  beforeEach(() => {
+    props = {
+      runId: 'mockId',
+      setSetupScreen: vi.fn(),
+    }
+    vi.mocked(useMostRecentCompletedAnalysis).mockReturnValue({
+      runTimeParameters: mockRunTimeParameterData,
+    } as any)
```

Comment:
Probably using vitest-when would be better.

```suggestion
import { when } from 'vitest-when'

const RIN_ID = 'mockId'
    when(vi.mocked(useMostRecentCompletedAnalysis)).calledWith(RUN_ID).thenReturn({
      runTimeParameters: mockRunTimeParameterData,
    } as any)
```